session/session.go

```diff
@@ -160,7 +161,9 @@ func (h *StmtHistory) Count() int {
 type session struct {
 	// processInfo is used by ShowProcess(), and should be modified atomically.
 	processInfo atomic.Value
-	txn         TxnState
+	// txns holds multiple txns for different txn scopes, which will be like:
+	// txn_scope -> TxnState
+	txns map[string]*TxnState
```

Comment:
Is there more than one transaction in a session at the same time?

Reply:
Yes, we will prepare at least two transactions in a session when we set the `txn_scope` to non-global, and usually one is global and another is local. But it's also possible to have more than two transactions if the user changes the session variable `txn_scope` to different values frequently. I'm planning to do some GC work to handle these useless transactions.